provider/resources.go

```diff
@@ -156,6 +159,65 @@ func boolRef(b bool) *bool {
 	return &b
 }
 
+// stringValue gets a string value from a property map if present, else ""
+func stringValue(vars resource.PropertyMap, prop resource.PropertyKey, envs []string) string {
+	val, ok := vars[prop]
+	if ok && val.IsString() {
+		return val.StringValue()
+	}
+	for _, env := range envs {
+		val, ok := os.LookupEnv(env)
+		if ok {
+			return val
+		}
+	}
+	return ""
+}
+
+func preConfigureCallback(vars resource.PropertyMap, c shim.ResourceConfig) error {
+
+	// explicitly check to make sure that the user has a project available before we do
+	// anything with the provider
+	project := stringValue(vars, "project", []string{"GOOGLE_PROJECT", "GOOGLE_CLOUD_PROJECT",
+		"GCLOUD_PROJECT",
+		"CLOUDSDK_CORE_PROJECT"})
+	if project == "" {
+		return fmt.Errorf("unable to find required configuration setting: GCP Project\n" +
+			"Set the GCP Project by using:\n" +
+			"\t`pulumi config set gcp:project <project>`")
+	}
+
+	// check if skipCredentialsValidation is True, if yes, exit early
+	if val, ok := vars["skipCredentialsValidation"]; ok {
+		if val.IsBool() {
+			return nil
+		}
```

Comment:
This doesn't look right to me. I think this will return true if any bool is set.

Reply:
It turns out to be unnecessary in the first place -thank you for catching!